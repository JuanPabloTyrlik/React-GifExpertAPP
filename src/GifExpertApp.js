import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';
import PropTypes from 'prop-types';

const GifExpertApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState(defaultCategories);
    // const [categories, setCategories] = useState([
    //     'Shingeki no Kyojin',
    //     'Kimetsu no Yaiba',
    //     'One-Punch Man',
    //     "Jojo's Bizarre Adventure",
    // ]);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />
            <ol>
                {categories.map((category) => (
                    <GifGrid key={category} category={category} />
                ))}
            </ol>
        </>
    );
};

GifExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}

export default GifExpertApp;
