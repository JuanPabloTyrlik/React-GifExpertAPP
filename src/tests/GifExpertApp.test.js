import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Tests on GifExpertApp', () => {
    test('should render correctly', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('should render a list of Categories', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        expect(wrapper).toMatchSnapshot();
    })
    
});
