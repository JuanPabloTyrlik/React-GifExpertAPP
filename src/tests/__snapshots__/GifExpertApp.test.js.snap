// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifExpertApp should render a list of Categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Tests on GifExpertApp should render correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
